src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -1001,3 +1003,73 @@ void TieredHNSWIndex<DataType, DistType>::TieredHNSW_BatchIterator::filter_irrel
     // Update number of results (pop the tail)
     array_pop_back_n(rl.results, end - cur_end);
 }
+
+template <typename DataType, typename DistType>
+VecSimIndexInfo TieredHNSWIndex<DataType, DistType>::info() const {
+    auto info = VecSimTieredIndex<DataType, DistType>::info();
+    info.tieredInfo.backendAlgo = VecSimAlgo_HNSWLIB;
+    info.tieredInfo.backendInfo.hnswInfo = this->backendIndex->info().hnswInfo;
+    HnswTieredInfo hnswTieredInfo = {.pendingSwapJobsThreshold = this->pendingSwapJobsThreshold};
+    info.tieredInfo.specificTieredBackendInfo.hnswTieredInfo = hnswTieredInfo;
+
+    return info;
+}
+
+template <typename DataType, typename DistType>
+VecSimInfoIterator *TieredHNSWIndex<DataType, DistType>::infoIterator() const {
```

Comment:
Why did these two functions move? it is almost entirely common to any tiered algorithm, Consider overriding and calling the ancestor's function for the extra stuff.

Reply:
For `info` that's basically what I did, I'll do it for `infoIterator` as well